tests/rpc/spec/features_spec.js

```diff
@@ -0,0 +1,109 @@
+describe('Features', function(){
+
+    function handleEvent(name, handler) {
+        channel.handleEvent(name, handler);
+            handlersToClean.push({
+            name: name,
+            handler: handler
+        });
+    };
+
+    beforeEach(function(done) {
+        channel.onReady(function() {
+             // Reset map and event counter.
+            channel.resetState(function() {
+                eventCounter = 0;
+                // Get default position
+                channel.getMapPosition(function(data) {
+                    defaultPosition = data;
+                    // update test feature coordinates
+                    pointGeojsonObject.features[0]. geometry.coordinates = [defaultPosition.centerX, defaultPosition.centerY];
+                    pointGeojsonObject.crs.properties.name = defaultPosition.srsName
+                    lineGeojsonObject.features[0]. geometry.coordinates = [[defaultPosition.centerX, defaultPosition.centerY], [defaultPosition.centerX + 10, defaultPosition.centerY+ 10]];
+                    lineGeojsonObject.crs.properties.name = defaultPosition.srsName
+                    // Delay between tests
+                    setTimeout(function() {
+                        done();
+                    }, 0);
+                });
+             });
+        });
+    });
+
+    afterEach(function() {
+        // Spy callback.
+        expect(eventCounter).toEqual(1, "Event count does not match");
+        // Reset event handlers.
+        while (handlersToClean.length) {

```

Comment:
This seems to be repeated in each spec. Maybe move the code to a common teardown function that can be called from each spec.

Reply:
Agreed, eventHandler() and resetEventhandler() needs to be defined only once.